components/PoeModal.tsx

```diff
@@ -0,0 +1,659 @@
+        );
+        return;
+      }
+
+      if (!!currentMetadata) {
+        state.p2pClient?.abortRequest(
+          message.id,
+          "There's already a pending request"
+        );
+        return;
+      }
+      setAddress(message.sourceAddress);
+      setCurrentMetadata([message.id, message.appMetadata]);
+
+      if (message.operationDetails.length === 0) {
+        await state.p2pClient?.sendError(
+          message.id,
+          "Request was empty",
+          BeaconErrorType.TRANSACTION_INVALID_ERROR
+        );
+        setTransactionError("Operations were empty");
+        return;
+      }
+
+      const version = state.contracts[message.sourceAddress].version;
+      setTransfers(
+        (
+          await Promise.all(
+            message.operationDetails.map(async detail => {
+              switch (detail.kind) {
+                case TezosOperationType.TRANSACTION:
+                  if (!!detail.parameters) {
+                    try {
+                      const contract = await state.connection.contract.at(
+                        detail.destination
+                      );
+
+                      if (
+                        !contract.entrypoints.entrypoints[
+                          detail.parameters.entrypoint
+                        ]
+                      ) {
+                        throw new Error(
+                          `'${detail.parameters.entrypoint}' is not a valid entrypoint for ${detail.destination}`
+                        );
+                      }
+
+                      const methodSchema = new Schema(
+                        contract.entrypoints.entrypoints[
+                          detail.parameters.entrypoint
+                        ]
+                      );
+
+                      const value = contract.methodsObject[
+                        detail.parameters.entrypoint
+                      ](
+                        methodSchema.Execute(detail.parameters.value)
+                      ).toTransferParams().parameter?.value;
+
+                      if (!value) return undefined;
+
+                      const param = emitMicheline(value as Expr);
+
+                      const parser = new Parser();
+
+                      const type = emitMicheline(
+                        parser.parseJSON(
+                          contract.entrypoints.entrypoints[
+                            detail.parameters.entrypoint
+                          ]
+                        ),
+                        {
+                          indent: "",
+                          newline: "",
+                        }
+                      );
+
+                      return {
+                        type: "contract",
+                        values: {
+                          lambda: generateExecuteContractMichelson(version, {
+                            address: detail.destination,
+                            amount: Number(detail.amount),
```

Comment:
need to check and throw error.
if `detail.amount` is 'a', UI shows `-`. 
When click confirm, UI will show user "Failed to create the transaction. Please try again later". 
It's incorrect message.